guide/src/free-threading.md

```diff
@@ -0,0 +1,127 @@
+# Supporting Free-Threaded CPython
+
+CPython 3.13 introduces an experimental build of CPython that does not rely on
+the global interpreter lock for thread safety. As of version 0.23, PyO3 also has
+preliminary support for building rust extensions for the free-threaded Python
+build and support for calling into free-threaded Python from Rust.
+
+The main benefit for supporting free-threaded Python is that it is no longer
+necessary to rely on rust parallelism to achieve concurrent speedups using
+PyO3. Instead, you can parallelise in Python using the
+[`threading`](https://docs.python.org/3/library/threading.html) module, and
+still expect to see see multicore speedups by exploiting threaded concurrency in
+Python, without any need to release the GIL. If you have ever needed to use
+`multiprocessing` to achieve a speedup for some algorithm written in Python,
+free-threading will likely allow the use of Python threads instead for the same
+workflow.
```

Comment:
Maybe we want to also talk a bit about what PyO3/Rust brings to the table here? Maybe about the "fearless concurrency" point, having `Send`/`Sync` allowing to safely build APIs that can be used in this new context?